ui/components/package.json

```diff
@@ -32,9 +32,11 @@
     "lint:fix": "eslint --fix src --ext .ts,.tsx"
   },
   "dependencies": {
+    "@codemirror/lang-json": "^6.0.1",
     "@fontsource/lato": "^4.5.10",
     "@mui/x-date-pickers": "^5.0.0-beta.1",
     "@perses-dev/core": "0.20.0",
+    "@uiw/react-codemirror": "^4.19.1",
```

Comment:
With `npm` installing peer dependencies by default now, this feels like it's gonna be a constant question. 😂  This seems all right to me for now.

Reply:
I vaaaguely recall you mentioning at some point that codemirror might need to be a singleton on the page although having trouble googling it beyond seeing a few github issues mentioning that codemirror sometimes yells at you about having multiple instances of some packages. If that is the case, I think a peer dependency would be a lot safer for consumers.

Reply:
Yeah, it could have some funky behavior. 

How do peerDependencies work with transitive packages? e.g. `@uiw/react-codemirror` has `codemirror` as a peerDependency. Does that make `codemirror` a peerDependency for our `component` package now?

Reply:
I _think_ so, but I'm not 100% sure without testing it. It may depend on the version of `yarn` or `npm` the consumer is using because that behavior has changed over time.

Reply:
Are you planning to leave this as a dependency for now and dig into if this needs to be a peer dep later OR would it make sense to adjust it in this PR? (No strong feelings from me, just wanted to clarify intent since this moved into a real PR).

Reply:
Yeah, exactly not sure what it needs to be but a dependency is probably makes the most sense right now. If this doesn't work properly, let's revisit